File: modules/cannops/samples/sample.py
# This file is part of OpenCV project.
# It is subject to the license terms in the LICENSE file found in the top-level directory
# of this distribution and at http://opencv.org/license.html.

import numpy as np
import cv2

img = cv2.imread("/path/to/img")

cv2.cann.initAcl()
cv2.cann.setDevice(0)

ascendMat = cv2.cann.AscendMat()
ascendMat.upload(img)

ascendMatSum = cv2.cann.add(ascendMat, ascendMat)
imgResult = ascendMatSum.download()
print(imgResult)

cv2.cann.finalizeAcl()

File: modules/cannops/samples/sample.cpp
// This file is part of OpenCV project.
// It is subject to the license terms in the LICENSE file found in the top-level directory
// of this distribution and at http://opencv.org/license.html.

// g++ -o sample sample.cpp -I opencv/include/opencv4/ -L opencv/build/install/lib/ -l
// opencv_cannops -l opencv_core -l opencv_imgcodecs

#include <iostream>
#include <opencv2/imgcodecs.hpp>
#include <opencv2/cann.hpp>
#include <opencv2/cann_interface.hpp>

int main()
{
    cv::Mat img = cv::imread("/path/to/img");

    cv::cann::initAcl();
    cv::cann::setDevice(0);

    cv::cann::AscendMat ascendMat;
    ascendMat.upload(img);

    cv::cann::AscendMat ascendMatSum;
    cv::cann::add(ascendMat, ascendMat, ascendMatSum);
    cv::Mat imgResult;
    ascendMatSum.download(imgResult);
    std::cout << imgResult << std::endl;

    cv::cann::resetDevice();
    cv::cann::finalizeAcl();

    return 0;
}
